faster_coco_eval/core/cocoeval.py

```diff
@@ -468,33 +478,64 @@ def _summarizeDets():
+                )  # AP_*label*
 
             if self.lvis_style:
-                stats[14] = self._summarize(1, maxDets=self.params.maxDets[-1], freq_group_idx=0)  # APr
-                stats[15] = self._summarize(1, maxDets=self.params.maxDets[-1], freq_group_idx=1)  # APc
-                stats[16] = self._summarize(1, maxDets=self.params.maxDets[-1], freq_group_idx=2)  # APf
-
-            stats[6] = self._summarize(0, maxDets=self.params.maxDets[0])  # AR_first or AR_all
+                start_idx = 2 * nb_rngs + 8
+                stats[start_idx] = self._summarize(
+                    1, maxDets=self.params.maxDets[-1], freq_group_idx=0
+                )  # APr
+                stats[start_idx + 1] = self._summarize(
+                    1, maxDets=self.params.maxDets[-1], freq_group_idx=1
+                )  # APc
+                stats[start_idx + 2] = self._summarize(
+                    1, maxDets=self.params.maxDets[-1], freq_group_idx=2
+                )  # APf
+
+            start_idx = 3 + nb_rngs
+            stats[start_idx] = self._summarize(
+                0, maxDets=self.params.maxDets[0]
+            )  # AR_first or AR_all
             if len(self.params.maxDets) >= 2:
-                stats[7] = self._summarize(0, maxDets=self.params.maxDets[1])  # AR_second
+                stats[start_idx + 1] = self._summarize(
+                    0, maxDets=self.params.maxDets[1]
+                )  # AR_second
             if len(self.params.maxDets) >= 3:
-                stats[8] = self._summarize(0, maxDets=self.params.maxDets[2])  # AR_third
-
-            stats[9] = self._summarize(0, areaRng="small", maxDets=self.params.maxDets[-1])  # AR_small
-            stats[10] = self._summarize(0, areaRng="medium", maxDets=self.params.maxDets[-1])  # AR_medium
-            stats[11] = self._summarize(0, areaRng="large", maxDets=self.params.maxDets[-1])  # AR_large
+                stats[start_idx + 2] = self._summarize(
+                    0, maxDets=self.params.maxDets[2]
+                )  # AR_third
 
-            stats[12] = self._summarize(0, iouThr=0.5, maxDets=self.params.maxDets[-1])  # AR_50
-            stats[13] = self._summarize(0, iouThr=0.75, maxDets=self.params.maxDets[-1])  # AR_75
+            start_idx = 3 + nb_rngs + 3
+            for idx, label in enumerate(self.params.areaRngLbl):
+                if label == "all":
+                    continue
+                stats[start_idx + idx - 1] = self._summarize(
+                    0, areaRng=label, maxDets=self.params.maxDets[-1]
+                )  # AR_*label*
+
+            start_idx = 3 + 2 * nb_rngs + 3
+            stats[start_idx] = self._summarize(
+                0, iouThr=0.5, maxDets=self.params.maxDets[-1]
+            )  # AR_50
+            stats[start_idx + 1] = self._summarize(
+                0, iouThr=0.75, maxDets=self.params.maxDets[-1]
+            )  # AR_75
 
             return stats
```

Comment:
How hard to understand =)
Can this be simplified somehow more intuitively?
So that it would be more similar to _summarizeKps for example.
I don't know whether to add comments or improve the formatting.

Reply:
yep!